src/util/side_metadata/global.rs

```diff
@@ -0,0 +1,778 @@
+use super::constants::*;
+use super::helpers::*;
+use crate::util::{constants, Address};
+use crate::util::{heap::layout::vm_layout_constants::BYTES_IN_CHUNK, memory};
+use std::sync::atomic::{AtomicU16, AtomicU32, AtomicU8, AtomicUsize, Ordering};
+
+#[derive(Clone, Copy)]
+pub enum SideMetadataScope {
+    Global,
+    PolicySpecific,
+}
+
+impl SideMetadataScope {
+    pub fn is_global(&self) -> bool {
+        matches!(self, SideMetadataScope::Global)
+    }
+}
+
+#[derive(Clone, Copy)]
+pub struct SideMetadataSpec {
```

Comment:
Probably add a comment to say that we expect each plan or each policy to create constants for this struct. Those have to be constants for performance.

Reply:
done